IDLE_DWELL/Find_Idle_Dwells.py

```diff
@@ -0,0 +1,256 @@
+
+    # ----------------------------------  BEGIN DWELL PROCESSING ---------------------------------------------------
+    # The assumption here is that at the start of a load, you will not be in the middle of
+    # any kind of observation (neither science nor ECS)
+    # So for the first check, set the start of the search be the first command,
+    # and look for the first Start Science.
+    
+    dwell_start_date = all_commands[0]["date"]
+    dwell_start_time = all_commands[0]["time"]
+    
+    # Now scan through all the rest of the extracted  commands until
+    # you see a start science. That defines the first "dwell"
+    index = 0
+
+    while index < len(sim_acis_commands):
+
+        # Is this an OORMPDS command? If so set the value of the
+        # state dictionary
+         if "OORMPDS" in  sim_acis_commands[index]["tlmsid_string"]:
+             state_dict["radmon_status"] = "RADMON_DIS"
+                   
+        # Is this an OORMPEN command? If so set the value of the
+        # state dictionary
+         if "OORMPEN" in  sim_acis_commands[index]["tlmsid_string"]:
+             state_dict["radmon_status"] = "RADMON_EN"
+        
+        # Is this a SIMTRANS? If so set the present instrument to the commanded value
+         if sim_acis_commands[index]["command_type"] == "SIMTRANS":
+            # If was_an_obs is False, set the previous FP instrument to the
+            # present FP instrument. When there was a previous observation,
+            # the processing for the AA00 command does this.
+            if state_dict["was_an_obs"] == False:
+                state_dict["previous_fp_instrument"] = state_dict["present_fp_instrument"]
+            # Now set the present FP instrument to this SIMTRANS command value
+            state_dict["present_fp_instrument"] = BFCI.Get_Instrument(sim_acis_commands[index]["tlmsid_string"])
+
+        # Is this a Start Science Command?
+         if ("XTZ0000005" in  sim_acis_commands[index]["tlmsid_string"]) or \
+           ("XCZ0000005" in  sim_acis_commands[index]["tlmsid_string"]):
+            # If yes, you have enough information to calculate the Dwell Time
+            dwell_stop_date = sim_acis_commands[index]["date"].lstrip().rstrip()
+            dwell_stop_time = sim_acis_commands[index]["time"]
+    
+            # Calculate the delta t
+            dwell_time_secs, dwell_time_mins, dwell_time_hours = cd.Calc_Delta_Time(dwell_start_date, dwell_stop_date)
+            
+            # If the instrument is ACIS-I or ACIS-S or we are in the radzone
+            # executing an ECS measurement, and the dwell time is 1.5 hours or more
+            # then create an entry for the long_dwell_list.
+            if ( (state_dict["present_fp_instrument"] == "ACIS-I") or \
+                 (state_dict["present_fp_instrument"] == "ACIS-S") or \
+                 (state_dict["radmon_status"] == "RADMON_DIS") ) and \
```

Comment:
I'm confused on two points here: (1) if there is a long dwell to a NIL, wouldn't we want that to be flagged?  I don't think it would show up.  (2) It looks to me like the delta during usual perigee passage between inbound-ECS to outbound ECS will result in a warning.

Am I missing something that would cull those? If not, to avoid (2) you could create a first_ECS_flag in the same way you do the first_AA00_flag, have it set to False at RADMON_EN and when you encounter a radmon_status==RADMON_DIS with  first_ECS_flag==False you would set first_ECS_flag=True.  Your boolean here would then become:
   state[inst]==ACIS-I or state[inst]==ACIS-S or (state[radmon]==RADMON_DIS and state[first_ECS_flag]==False)) and dwelltime >= 1.5hr  
and would avoid the warning for perigee passages (except I guess for the recaptured science cases).